tests/tests/tier0/monitor-node-reconnect/test_monitor_node_reconnect.py

```diff
@@ -0,0 +1,50 @@
+# SPDX-License-Identifier: GPL-2.0-or-later
+
+import os
+import time
+import pytest
+from typing import Dict
+
+from bluechi_test.test import BluechiTest
+from bluechi_test.container import BluechiControllerContainer, BluechiNodeContainer
+from bluechi_test.config import BluechiControllerConfig, BluechiNodeConfig
+
+
+node_name = "node-foo"
+
+
+def exec(ctrl: BluechiControllerContainer, nodes: Dict[str, BluechiNodeContainer]):
+    result, output = ctrl.run_python(os.path.join("python", "is_node_connected.py"))
+    if result != 0:
+        raise Exception(output)
+
+    ctrl.exec_run("systemctl stop bluechi")
+    _, output = ctrl.exec_run('systemctl is-active bluechi')
+    assert output == 'inactive'
+
+    ctrl.exec_run("systemctl start bluechi")
+    ctrl.wait_for_bluechi()
+    # wait a short while for bluechi-agent to reconnect
+    time.sleep(1)
```

Comment:
Good point! 
Yes and no. Even though `bluechi` might be up, the `bluechi-agent` might not have connected directly since the heartbeat interval is 500ms. The 1s here is to make sure this test doesn't become flaky. I'll extend the inline comment a bit. Thanks!